common/djangoapps/student/helpers.py

```diff
@@ -35,18 +60,79 @@ def set_logged_in_cookie(request, response):
         expires_time = time.time() + max_age
         expires = cookie_date(expires_time)
 
-    response.set_cookie(
-        settings.EDXMKTG_COOKIE_NAME, 'true', max_age=max_age,
-        expires=expires, domain=settings.SESSION_COOKIE_DOMAIN,
-        path='/', secure=None, httponly=None,
-    )
+    cookie_settings = {
+        'max_age': max_age,
+        'expires': expires,
+        'domain': settings.SESSION_COOKIE_DOMAIN,
+        'path': '/',
+        'secure': None,
+        'httponly': None,
+    }
+
+    # Backwards compatibility: set the cookie indicating that the user
+    # is logged in.  This is just a boolean value, so it's not very useful.
+    # In the future, we should be able to replace this with the "user info"
+    # cookie set below.
+    response.set_cookie(settings.EDXMKTG_LOGGED_IN_COOKIE_NAME, 'true', **cookie_settings)
+
+    # Set a cookie with user info.  This can be used by external sites
+    # to customize content based on user information.  Currently,
+    # we include information that's used to customize the "account"
+    # links in the header of subdomain sites (such as the marketing site).
+    if user is not None:
+        header_urls = {'logout': reverse('logout')}
+
+        # Unfortunately, this app is currently used by both the LMS and Studio login pages.
+        # If we're in Studio, we won't be able to reverse the account/profile URLs.
+        # To handle this, we don't add the URLs if we can't reverse them.
+        # External sites will need to have fallback mechanisms to handle this case
+        # (most likely just hiding the links).
```

Comment:
@AlasdairSwan This is something the marketing site JavaScript will need to handle gracefully.  We can't construct the URLs correctly if the user logs in through Studio, so they won't be included in the cookie.